zerver/views/auth.py

```diff
@@ -80,6 +80,52 @@ def redirect_to_subdomain_login_url():
     redirect_url = login_url + '?subdomain=1'
     return HttpResponseRedirect(redirect_url)
 
+def login_or_ask_for_registration(request, email, user_profile, full_name='',
+                                  invalid_subdomain=False):
+    # type: (HttpRequest, Text, UserProfile, Text, Optional[bool]) -> HttpResponse
+    if invalid_subdomain:
+        # Show login page with an error message
+        return redirect_to_subdomain_login_url()
+
+    if user_profile is None:
```

Comment:
This should check `or user_profile.is_mirror_dummy:`, like `login_or_regtister_remote_user`, right?